deps/lua/src/ldo.c

```diff
@@ -226,6 +226,7 @@ static StkId adjust_varargs (lua_State *L, Proto *p, int actual) {
   }
 #endif
   /* move fixed parameters to final position */
+  luaD_checkstack(L, p->maxstacksize);  /*   check again for new 'base' */
```

Comment:
From analyzing the code, I am not fully sure if this `luaD_checkstack` is really required because it is done from outside on `luaD_precall:277`. Notice that on Lua 5.1, it makes sure to check for `p->maxstacksize + p->numparams` which is enough for anything it will potentially need.